tests/csapi/apidoc_room_history_visibility_test.go

```diff
@@ -0,0 +1,337 @@
+package csapi_tests
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/tidwall/gjson"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/client"
+	"github.com/matrix-org/complement/internal/match"
+	"github.com/matrix-org/complement/internal/must"
+	"github.com/matrix-org/complement/runtime"
+)
+
+// TODO most of this can be refactored into data-driven tests
+
+func fetchEvent(t *testing.T, c *client.CSAPI, roomId, eventId string) *http.Response {
+	return c.DoFunc(t, "GET", []string{"_matrix", "client", "r0", "rooms", roomId, "event", eventId})
+}
+
+func createRoomWithVisibility(t *testing.T, c *client.CSAPI, visibility string) string {
+	return c.CreateRoom(t, map[string]interface{}{
+		"initial_state": []map[string]interface{}{
+			{
+				"content": map[string]interface{}{
+					"history_visibility": visibility,
+				},
+				"type": "m.room.history_visibility",
```

Comment:
Suprised this works without a `state_key` explicitly set. Can we do this please?